src/ray/raylet/node_manager.cc

```diff
@@ -1814,9 +1814,9 @@ bool NodeManager::AssignTask(const Task &task) {
   cluster_resource_map_[my_client_id].Acquire(spec.GetRequiredResources());
 
   if (spec.IsActorCreationTask()) {
-    // Check that we are not placing an actor creation task on a node with 0 CPUs.
-    RAY_CHECK(cluster_resource_map_[my_client_id].GetTotalResources().GetResourceMap().at(
-                  kCPU_ResourceLabel) != 0);
+    // Check that the actor's placement resource requirements are satisfied.
+    RAY_CHECK(spec.GetRequiredPlacementResources().IsSubset(
+        cluster_resource_map_[my_client_id].GetTotalResources()));
```

Comment:
this change looks like it would allow placing zero CPU actor creation tasks on a node with zero CPUs.  The current behavior is to disallow it. Is this what we want?

Reply:
Well, the current behavior is to crash according to https://github.com/ray-project/ray/issues/4892. I view this more as a bug fix and not a change in behavior. The requirement of needing a CPU for placing the actor only applies to actors that do not acquire lifetime resources. The actors that do acquire lifetime resources can be placed as long as they can acquire those resources.